clang/lib/AST/ItaniumMangle.cpp

```diff
@@ -704,6 +704,15 @@ ItaniumMangleContextImpl::getEffectiveDeclContext(const Decl *D) {
       return D->getLexicalDeclContext()->getRedeclContext();
   }
 
+  if (const auto *FTD = dyn_cast<FunctionTemplateDecl>(D)) {
+    // Member-like constrained friends are mangled as if they were members of
```

Comment:
Let's reuse the function decl above.

We could use the `Decl::getAsFunction()` to do both of these checks for us.
```cpp
if (const auto *FD = D->getAsFunction())
```

Reply:
Great idea!